superset-frontend/src/views/CRUD/alert/ExecutionLog.tsx

```diff
@@ -0,0 +1,156 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { styled, t } from '@superset-ui/core';
+import moment from 'moment';
+import React, { useMemo } from 'react';
+import { Link, useParams } from 'react-router-dom';
+import ListView from 'src/components/ListView';
+import SubMenu from 'src/components/Menu/SubMenu';
+import withToasts from 'src/messageToasts/enhancers/withToasts';
+import { fDuration } from 'src/modules/dates';
+import AlertStatusIcon from 'src/views/CRUD/alert/components/AlertStatusIcon';
+import {
+  useListViewResource,
+  useSingleViewResource,
+} from 'src/views/CRUD/hooks';
+import { AlertObject, LogObject } from './types';
+
+const PAGE_SIZE = 25;
+
+const StyledHeader = styled.div`
+  display: flex;
+  flex-direction: row;
+
+  a,
+  Link {
+    margin-left: 16px;
+    font-size: 12px;
+    font-weight: normal;
+    text-decoration: underline;
+  }
+`;
+
+interface ExecutionLogProps {
+  addDangerToast: (msg: string) => void;
+  addSuccessToast: (msg: string) => void;
+  isReportEnabled: boolean;
+}
+
+function ExecutionLog({ addDangerToast, isReportEnabled }: ExecutionLogProps) {
+  const { alertId }: any = useParams();
+  const {
+    state: { loading, resourceCount: logCount, resourceCollection: logs },
+    fetchData,
+  } = useListViewResource<LogObject>(
+    `report/${alertId}/log`,
+    t('log'),
+    addDangerToast,
+    false,
+  );
+  const {
+    state: { loading: alertLoading, resource: alertResource },
+    fetchResource,
+  } = useSingleViewResource<AlertObject>(
+    'report',
+    t('reports'),
+    addDangerToast,
+  );
+
+  if (alertId !== null && !alertLoading) {
+    fetchResource(alertId);
+  }
```

Comment:
I haven't tested this, but it looks like this would cause an infinite fetch loop. Let's wrap it in a useEffect to be safe
```suggestion
  useEffect(() => {
    if (alertId !== null) {
      fetchResource(alertId);
    }
  }, [alertId])

```

Reply:
yup, just tested this. It results in an infinite `fetchResource` loop